src/middleware.ts

```diff
@@ -248,7 +283,15 @@ export const persist = <S extends State>(
       blacklist.forEach((key) => delete state[key])
     }
 
-    return storage?.setItem(name, await serialize({ state, version }))
+    return thenableSerialize({ state, version }).then((serializedValue) => {
+      if (storage) {
+        storage.setItem(name, serializedValue)
```

Comment:
As we don't return the `storage.setItem` result, `return` in L286 is useless.
We don't use the return value of setItem anywhere (or did I miss it?), could we remove `return` in L286? (and that will change other code, if it works.)

Reply:
L338 uses it.
So, we should return thenable.
Please define `thenableStorageSetItem` in advance, and return here. keep `return` in L286.

Reply:
should I define `thenableStorageSetItem` as a function ?

Reply:
Yeah, like you did with getItem.
```js
const thenableStorageSetItem = toThenable(storage.setItem.bind(storage))
```
Defining it in advance is better because it will be invoked multiple times.

Reply:
in `line:288` we should also return `storage.setItem(...)` so if it threw an error the `catch` block `in line:347` can handle it.

Reply:
I think `thenableStorageSetItem()` that you mentioned @dai-shi is not possible with current `toThenable` solution. because if we return `thenableStorageSetItem(...)` in `line:288` then the `toThenable` will wrap the `thenableStorageSeteItem` with another `thenable`.

Reply:
dear @dai-shi please have a comment on this.

Reply:
Sorry, I think this was mymisundersting.
```ts
  return (storage as StateStorage).setItem(name, serializedValue)
```
just works right? (apart from the typing issue)